rfcs/0158-artifact-metadata.md

```diff
@@ -0,0 +1,69 @@
+# RFC 158 - artifact metadata
+* Comments: [#158](https://github.com/taskcluster/taskcluster-rfcs/pull/158)
+* Proposed by: @escapewindow
+
+# Summary
+
+The goal is to provide Artifact Integrity guarantees from the point the worker uploads an artifact, to the point that someone downloads that artifact to use. We can do this by:
+
+1. adding SHA metadata to artifacts in the Queue,
+2. ensuring that that metadata can't be modified once it's written, and
+3. providing a download tool that queries the Queue for an artifact's location and SHA, downloads the artifact, and verifies the downloaded SHA matches the SHA provided by the Queue.
+
+## Motivation
+
+This would improve robustness and security. By storing a SHA to verify on download, we can avoid corrupt downloads. By verifying the read-only SHA before we use the artifact, we can detect malicious tampering of artifacts at rest before we use it for a critical operation.
+
+(This would obsolete the [artifacts section of the Chain of Trust artifact](https://scriptworker.readthedocs.io/en/latest/chain_of_trust.html#chain-of-trust-artifacts), and improve artifact integrity platform-wide. See the [Chain of Trust implications](#chain-of-trust-implications) section below.)
+
+# Details
+
+## Adding artifact metadata to the Queue
+
+First, we add a `metadata` dictionary to the `S3ArtifactRequest` type. This is a dictionary to allow for flexibility of usage. The initial known keys should include
+
+```
+ContentFilesize int64 `json:"contentFilesize"`
+ContentSha256 string `json:"contentSha256"`
+```
+
+The sha256 field is required for Artifact Integrity. The filesize field is optional.
+
+A future entry may be `ContentSha256WorkerSignature`, if we solve worker identity, though we may abandon this in favor of worker [re]registration.
```

Comment:
What does "if we solve worker identity" mean?

Reply:
This is referring to https://github.com/taskcluster/taskcluster-rfcs/issues/157 , which is looking less likely to happen.